src/test/java/com/nikolenko/homeworks/homework_24/DBDataGetterTest.java

```diff
@@ -0,0 +1,176 @@
+        assertTrue(result.contains("Hanks"));
+        assertFalse(result.contains("Sheep"));
+    }
+
+    @Test
+    void punkt4() {
+        /*Получить список всех работников у которых последняя буква в имени 'd'*/
+        String query = "SELECT * FROM EMPLOYEES where FIRST_NAME like '%d'";
+        String result = runEmployeeRequest(query);
+        log.info(result);
+        assertTrue(result.contains("David"));
+        assertTrue(result.contains("Shmavid"));
+        assertFalse(result.contains("Dolly"));
+    }
+
+    @Test
+    void punkt5() {
+        /*Получить список всех работников у которых зарплата находится в промежутке от 2000 до 4000 (включительно)*/
+        String query = "SELECT * FROM EMPLOYEES where SALARY >= 2000 and SALARY <= 4000";
+        String result = runEmployeeRequest(query);
+        log.info(result);
+        assertTrue(result.contains("Grass"));
+        assertTrue(result.contains("Sawier"));
+        assertFalse(result.contains("Bush"));
+        assertFalse(result.contains("Sheep"));
+    }
+
+    @Test
+    void punkt6() {
+        /*Получить список работников которые живут в Europe (region_name)*/
+        String query = "SELECT EMPLOYEE_ID, FIRST_NAME, LAST_NAME, SALARY" +
+                " FROM EMPLOYEES em, DEPARTMENTS dp, LOCATIONS lc, COUNTRIES cn, REGIONS rg" +
+                " WHERE em.DEPARTMENT_ID=dp.DEPARTMENT_ID" +
+                " and dp.LOCATION_ID = lc.LOCATION_ID" +
+                " and lc.COUNTRY_ID = cn.COUNTRY_ID" +
+                " and cn.REGION_ID = rg.REGION_ID" +
+                " and rg.REGION_NAME = 'EUROPE'";
+
+
+        String result = runEmployeeRequest(query);
+        log.info(result);
+        assertTrue(result.contains("Lessley"));
+        assertTrue(result.contains("Dave"));
+        assertFalse(result.contains("Bush"));
+        assertFalse(result.contains("Sawier"));
+    }
+
+    @Test
+    void punkt7() {
+        /*Показать всех менеджеров которые имеют в подчинении больше 6ти работников*/
+        String query = "SELECT EMPLOYEE_ID, FIRST_NAME, LAST_NAME, SALARY " +
+                "FROM Employees em, (SELECT COUNT(EMPLOYEE_ID), MANAGER_ID FROM Employees GROUP BY MANAGER_ID HAVING COUNT(EMPLOYEE_ID) > 6) mn" +
+                " WHERE mn.MANAGER_ID = em.EMPLOYEE_ID";
+
+        String result = runEmployeeRequest(query);
+        log.info(result);
+        assertTrue(result.contains("Dave"));
+        assertTrue(result.contains("Sheep"));
+        assertFalse(result.contains("Bush"));
+        assertFalse(result.contains("Sawier"));
+    }
+
+    @Test
+    void punkt8() {
+        /*Получить список работников у которых менеджер получает зарплату больше 10000*/
+        String query = "SELECT * FROM EMPLOYEES empls, (SELECT EMPLOYEE_ID, SALARY FROM EMPLOYEES em, (SELECT MANAGER_ID as mngr FROM Employees GROUP BY MANAGER_ID) mngrs WHERE EMPLOYEE_ID = mngr AND SALARY >10000) richmngr WHERE empls.MANAGER_ID = richmngr.EMPLOYEE_ID";
```

Comment:
SELECT *
  FROM employees
 WHERE manager_id IN (SELECT employee_id
                        FROM employees
                       WHERE salary > 10000)